server/game/cards/characters/02/croneofvaesdothrak.js

```diff
@@ -5,16 +5,16 @@ class CroneOfVaesDothrak extends DrawCard {
     setupCardAbilities(ability) {
         this.reaction({
             when: {
-                onCardDiscarded: (event, player, card, allowSave, originalLocation) =>
-                    ((originalLocation === 'hand' || originalLocation === 'draw deck')
-                     && card.getType() === 'character'
-                     && player !== this.controller)
+                onCardDiscarded: (event, params) =>
```

Comment:
I agree, it'd be way better if it were all just on the event object (which is where I wanted to move ultimately anyway). Let me take a crack at that here, though it'll probably require a new variant of `raiseEvent`.